haystack/components/tools/tool_invoker.py

```diff
@@ -504,6 +504,9 @@ def run(
                     streaming_callback(
                         StreamingChunk(
                             content="",
+                            index=len(tool_messages) - 1,
+                            tool_call_result=tool_messages[-1].tool_call_results[0],
+                            start=True,
```

Comment:
In ToolInvoker the whole ToolCallResult is always sent so start is always True and index is just the current message index.